api/src/test/java/org/apache/iceberg/expressions/TestAggregateEvaluator.java

```diff
@@ -0,0 +1,184 @@
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), -33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 3333)));
+
+  private static final DataFile[] dataFiles = {
+    FILE, MISSING_SOME_NULLS_STATS_1, MISSING_SOME_NULLS_STATS_2
+  };
+
+  @Test
+  public void testIntAggregate() {
+    List<Expression> list =
+        ImmutableList.of(
+            Expressions.countStar(),
+            Expressions.count("id"),
+            Expressions.max("id"),
+            Expressions.min("id"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    Assert.assertTrue(aggregateEvaluator.allAggregatorsValid());
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {90L, 60L, 3333, -33};
+    assertEvaluatorResult(result, expected);
+  }
+
+  @Test
+  public void testAllNulls() {
+    List<Expression> list =
+        ImmutableList.of(
+            Expressions.countStar(),
+            Expressions.count("all_nulls"),
+            Expressions.max("all_nulls"),
+            Expressions.min("all_nulls"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    Assert.assertTrue(aggregateEvaluator.allAggregatorsValid());
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {90L, 0L, null, null};
+    assertEvaluatorResult(result, expected);
+  }
+
+  @Test
+  public void testSomeNulls() {
+    List<Expression> list =
+        ImmutableList.of(
+            Expressions.countStar(),
+            Expressions.count("some_nulls"),
+            Expressions.max("some_nulls"),
+            Expressions.min("some_nulls"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    Assert.assertFalse(aggregateEvaluator.allAggregatorsValid());
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {90L, null, null, null};
+    assertEvaluatorResult(result, expected);
+  }
+
+  @Test
+  public void testNoStats() {
+    List<Expression> list =
+        ImmutableList.of(
+            Expressions.countStar(),
+            Expressions.count("no_stats"),
+            Expressions.max("no_stats"),
+            Expressions.min("no_stats"));
+    AggregateEvaluator aggregateEvaluator = AggregateEvaluator.create(SCHEMA, list);
+    for (DataFile dataFile : dataFiles) {
+      aggregateEvaluator.update(dataFile);
+    }
+
+    Assert.assertFalse(aggregateEvaluator.allAggregatorsValid());
+    StructLike result = aggregateEvaluator.result();
+    Object[] expected = {90L, null, null, null};
```

Comment:
`"no_stats"` doesn't have stats except records counts
`count(*)` has values for all the FILES. It's 90L

`count("no_stats")` doesn't have value for all the FILEs, it became invalid in the first loop and we stop looking this. The value is null.

`max(("no_stats")` doesn't have value for all the FILEs, it became invalid in the first loop and we stop looking this. The value is null.

`min(("no_stats")` doesn't have value for all the FILEs, it became invalid in the first loop and we stop looking this. The value is null.